### PR TITLE
Importer

### PR DESCRIPTION
**Pull request by alxndrsn**
@garethbowen is this an acceptable approach for converting csv to json?  Currently implemented for contacts (people and places), but not reports.

### CODE REVIEW

File: src/fn/csv-to-docs.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        const doc = { type:contactType };
+
+        function addReference(r) {
+          r.doc = doc;
+          model.references.push(r);
+        }
+
+        for(let i=0; i<cols.length; ++i) {
+          const { col, val, references } = parseColumn(cols[i], r[i]);
+          doc[col] = val;
```
Comment (garethbowen) [inline src/fn/csv-to-docs.js]: We'll need to handle nested fields at some point, eg: `fields.name:Alex` => `{ fields: { name: "Alex" } }`

File: src/fn/upload-docs.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  const process = sets.reduce((promiseChain, docSet) =>
+    promiseChain.then(() =>
+      db.bulkDocs(docSet.map(fs.readJson))
+        .then(res => {
+          trace('Uploaded', docSet);
+          res.forEach(r => {
+            if(r.error) {
+              results.failed[r.id] = r.error;
+            } else {
+              results.ok.push(r.id);
```
Comment (garethbowen) [inline src/fn/upload-docs.js]: It'd be good to delete, or maybe just change the extension, of successfully imported docs. This means if it fails and you rerun it it won't try and upload the same docs again and get conflicts.

File: src/fn/csv-to-docs.js (hunk 1 above)
Comment (alxndrsn) [reply src/fn/csv-to-docs.js]: Is this done in contacts, or just reports?

File: src/fn/upload-docs.js (hunk 2 above)
Comment (alxndrsn) [reply src/fn/upload-docs.js]: Nice idea.

**Comment by alxndrsn** [comment]
Should be good to go now.

File: src/fn/csv-to-docs.js (hunk 1 above)
Comment (garethbowen) [reply src/fn/csv-to-docs.js]: Probably just reports, but I can't say for sure.

**Pull request closed without merging**